presto-orc/src/main/java/com/facebook/presto/orc/OrcOutputBuffer.java

```diff
@@ -269,18 +287,32 @@ public void writeBytes(byte[] source)
     }
 
     @Override
-    public void writeBytes(byte[] source, int sourceIndex, int length)
+    public void writeBytes(byte[] bytes, int bytesOffset, int length)
     {
-        // Write huge chunks direct to OutputStream
-        if (length >= DIRECT_FLUSH_SIZE) {
-            // todo fill buffer before flushing
+        // finish filling the buffer
+        if (bufferPosition != 0) {
```

Comment:
This code is almost exact copy/paste of the byte[]. This code is not performance critical (not called per input value). So can we refactor the code to remove duplication ?

Can we also remove DIRECT_FLUSH_SIZE concept completely and always fill to the max buffer sizes.

Reply:
Done. I had to rewrite ensureBatchSize to get rid of DIRECT_FLUSH_SIZE.

Reply:
Vader found an issue with a special case of Slices.EMPTY_SLICE which has null buffer array - fixed.

After that:
Verifier - ok
Vader - ok